x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/read_signals.ts

```diff
@@ -4,15 +4,63 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { AlertsClient } from '../../../../../alerting/server/alerts_client';
+import { findSignals } from './find_signals';
+import { SignalAlertType, isAlertTypeArray, ReadSignalParams } from './types';
 
-export interface ReadSignalParams {
-  alertsClient: AlertsClient;
-  id: string;
-}
+export const findSignalInArrayById = (objects: object[], id: string): SignalAlertType | null => {
+  if (isAlertTypeArray(objects)) {
+    const signals: SignalAlertType[] = objects;
+    const signal: SignalAlertType[] = signals.filter(datum => {
+      return datum.alertTypeParams.id === id;
+    });
+    if (signal.length !== 0) {
+      return signal[0];
+    } else {
+      return null;
+    }
```

Comment:
nitpicking but just from a readability standpoint - I think the logic would read cleaner if it was swapped

```
if (signal.length === 0) {
    return null;
} else {
    return signal[0];
}
```